src/service/xhr-impl.js

```diff
@@ -125,43 +122,43 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
+    const ampdocSingle = this.ampdocService.getAmpDoc();
     if (!getMode().test &&
-        this.ampdocSingle_ &&
+        ampdocSingle &&
         Math.random() < 0.01 &&
-        parseUrl(input).origin != this.win.location.origin &&
-        !Services.viewerForDoc(this.ampdocSingle_).hasBeenVisible()) {
-      dev().error('XHR', 'attempted to fetch %s before viewer was visible',
+        parseUrl(input).origin != this.win.location.origin) {
+      dev().error('XHR', 'attempted to fetch URL from different origin',
           input);
     }
```

Comment:
This entire if statement can be removed, since we're now waiting for viewer visible regardless.